app/collections/queries/getCollection.ts

```diff
@@ -0,0 +1,117 @@
+import { resolver, Ctx, AuthorizationError } from "blitz"
+import db from "db"
+
+export default resolver.pipe(async (suffix: string, ctx: Ctx) => {
+  const collection = await db.collection.findFirst({
+    where: {
+      suffix,
+    },
+    include: {
+      submissions: {
+        orderBy: {
+          updatedAt: "desc",
+        },
+        include: {
+          submittedBy: true,
+          module: true,
+          editor: {
+            include: {
+              workspace: true,
+            },
+          },
+        },
+      },
+      editors: {
+        orderBy: {
+          id: "asc",
+        },
+        include: {
+          workspace: {
+            include: {
+              members: {
+                include: {
+                  user: {
+                    select: {
+                      id: true,
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+      type: true,
+    },
+  })
+
+  const pendingSubmissions = await db.submission.findMany({
+    where: {
+      collectionId: collection!.id,
+      accepted: null,
+    },
+  })
+  const contributors = await db.submission.findMany({
+    where: {
+      accepted: true,
+      collectionId: collection!.id,
+      NOT: {
+        submittedBy: null,
+      },
+    },
+    select: {
+      submittedBy: true,
+    },
+  })
+  const uniqContributors = contributors.filter((value, index) => {
+    const _value = JSON.stringify(value)
+    return (
+      index ===
+      contributors.findIndex((obj) => {
+        return JSON.stringify(obj) === _value
+      })
+    )
+  })
+
+  const isFollowing = await db.collection.findFirst({
+    where: {
+      id: collection?.id,
+    },
+    include: {
+      followers: {
+        where: {
+          id: ctx.session.$publicData.workspaceId,
+        },
+      },
+    },
+  })
+
+  let isAdmin = false
+  if (ctx.session.$publicData.workspaceId) {
+    console.log("yes")
```

Comment:
I'm just double-checking that this logging is intended! If so, would it be useful to make it a more informative? 
(e.g., `console.log("Workspace ID is present")` etc)?

Reply:
lol no [chesterton's fence](https://fs.blog/chestertons-fence/) here! removed :)

Reply:
Didn't know about this—cool